content/general/queues.textile

```diff
@@ -0,0 +1,17 @@
+---
+title: Message Queues
+section: general
+index: 19
+---
+
+Message queues provide a reliable mechanism for customers to process, store, augment or reroute realtime data efficiently and asynchronously by workers.
+
+Using configurable rules, you can ensure that your messages, presence and channel lifecycle events are enqueued on an Ably message queue. Unlike our "channels":/realtime/channels-messages#channels which follow a "pub/sub pattern":https://en.wikipedia.org/wiki/Publish%E2%80%93subscribe_pattern where each message is delivered to any number of subscribers, our "message queues":https://en.wikipedia.org/wiki/Message_queue operate on the basis that each messages is delivered to only one subscriber.
+
+As each message is delivered exactly once, queues are commonly used by one or more workers to process realtime data published by Ably asynchronously. For example, using workers subscribed to a queue, you could persist each message of a live chat to your own database, start publishing updates once a channel becomes active, or trigger an event if a device has submitted a location that indicates that it has reached its destination.
```

Comment:
I would caution against using the expression "exactly once" except in situations where we state very precisely what assurances are provided by the system; people will be likely to infer things that we won't be able to guarantee.. Instead we use more explanatory language and talk about  "message delivery to a single subscriber" or "messages being distributed so each is delivered to a single consumer".